### PR TITLE
Narrow down allowed attribute vector data types for some feature executors

### PR DESCRIPTION
**Pull request by toregge**
@geirst, please review
@havardpe, FYI

### CODE REVIEW

File: searchlib/src/tests/features/prod_features_attributematch.cpp [hunk 1]
```diff
@@ -303,4 +304,13 @@ Test::testAttributeMatch()
             addScore("attributeMatch(wint).fieldCompleteness", 0);
         ASSERT_TRUE(ft.execute(exp));
     }
+    { // tensor attribute is not allowed
+        FtFeatureTest ft(_factory, "attributeMatch(tensor)");
+        ft.getIndexEnv().getBuilder().addField(FieldType::ATTRIBUTE, CollectionType::SINGLE, DataType::TENSOR, "tensor");
+        AttributePtr tensorAttr = AttributeFactory::createAttribute("tensor", AVC (AVBT::TENSOR, AVCT::SINGLE));
```
Comment (havardpe) [inline searchlib/src/tests/features/prod_features_attributematch.cpp]: I do not think we need an actual attribute vector for checking validation only

**Review by geirst — COMMENTED** [review]
LGTM. By the way, are there any more instances of visitDumpFeatures() that should be updated as well. If so, please add.

**Comment by toregge** [comment]
I'm not aware of any further needed changes to instances of visitDumpFeatures().

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**